web/src/main/java/it/pagopa/selfcare/party/registry_proxy/web/model/mapper/NationalRegistriesMapper.java

```diff
@@ -12,6 +13,9 @@ private NationalRegistriesMapper() {
     }
 
     public static LegalAddressResponse toResource(LegalAddressProfessionalResponse response) {
+        if(!StringUtils.hasText(response.getAddress())) {
+            throw new ResourceNotFoundException("Legal Address not found");
+        }
```

Comment:
you could throw ResourceNotFoundException by core services, es. nationalRegistresServices....business logic should not be inside mapper